File: Data Structures & Algorithms/jump-game/submission-0.cpp
class Solution {
public:
   static bool canJump(const std::vector<int>& nums) {
        int maxReach = 0;
        for (size_t i = 0; i < nums.size(); ++i) {
            maxReach = std::max(maxReach, nums[i]);
            if (maxReach == 0 && i < nums.size() - 1) {
                return false;
            }
            --maxReach;
        }
        return true;
    }
};

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        st = set()
        ans, ctr, prevNum = 0, 0, nums[0]
        for num in nums:
            st.add(num)
        while len(st) > 0:
            if prevNum in st:
                ctr += 1
                st.remove(prevNum)
            else:
                prevNum = st.pop()
                ctr = 1
            ans = max(ans, ctr)
            prevNum += 1
        return ans
